src/frontend/src/tests/lib/api/idb.api.spec.ts

```diff
@@ -0,0 +1,196 @@
+import {
+	deleteIdbBtcAddressMainnet,
+	deleteIdbEthAddress,
+	deleteIdbSolAddressMainnet,
+	getIdbBtcAddressMainnet,
+	getIdbEthAddress,
+	getIdbSolAddressMainnet,
+	setIdbBtcAddressMainnet,
+	setIdbEthAddress,
+	setIdbSolAddressMainnet,
+	updateIdbBtcAddressMainnetLastUsage,
+	updateIdbEthAddressLastUsage,
+	updateIdbSolAddressMainnetLastUsage
+} from '$lib/api/idb.api';
+import { Principal } from '@dfinity/principal';
+import * as idbKeyval from 'idb-keyval';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+
+vi.mock('idb-keyval', () => ({
+	createStore: vi.fn(() => ({
+		/* mock store implementation */
+	})),
+	set: vi.fn(),
+	get: vi.fn(),
+	del: vi.fn(),
+	update: vi.fn()
+}));
+
+vi.mock('$app/environment', () => ({
+	browser: true
+}));
+
+describe('idb.api', () => {
+	const mockPrincipal = Principal.fromText('2vxsx-fae');
```

Comment:
nitpick: there is already a reusable `mockPrincipal` defined in `src/frontend/src/tests/mocks/identity.mock.ts`